Remove duplicate routes with Array#uniq instead of Array#delete_if

vaihtovirta opened the pull request:
Hello!
Here is my little story behind this PR. :smile: 
#### Problem

Recently I've updated ruby to version 2.3.0. Then I've found that route helpers are no longer available in my application.
After quick investigation I've discovered that there is a problem with [GrapeRouteHelpers::AllRoutes#all_routes](https://github.com/reprah/grape-route-helpers/blob/master/lib/grape-route-helpers/all_routes.rb#L11) method, which returns empty array instead of array of **Grape::Route** objects.
The possible reason is that **Array#delete_if** behavior has [changed in ruby-2.3.0](https://github.com/ruby/ruby/blob/v2_3_0/NEWS#L170):

> Array#select!, Array#keep_if, Array#reject!, and Array#delete_if no longer changes the receiver array instantly every time the block is called.
#### Possible solution

The main idea is to collect array of route options and use **Array#uniq** against this array. Then we can pass only route options hash to [GrapeRouteHelpers::DecoratedRoute](https://github.com/vaihtovirta/grape-route-helpers/blob/refactor-deleting-duplicate-routes/lib/grape-route-helpers/decorated_route.rb#L12) instead of passing the whole **Grape::Route** instance.
I've checked, this solution gave me a consistent result on ruby-2.2.3 and ruby-2.3.0.

Please check out my PR and tell me what do you think.
I've also fixed specs and added rspec config for randomize specs order.

Thank you.

vaihtovirta committed:
Refactor removing duplicate routes
- Add uniq array of Grape::Route@options
- Get route's custom path from options[:as] instead of
- Rename all_routes.rb spec to all_routes_spec.rb so it'll run automatically
- Fix all_routes_spec
- Update decorated_route_spec according to changes
- Add rspec config for randomize spec order

timurkhafizov commented on lib/grape-route-helpers/decorated_route.rb:
```diff
@@ -1,16 +1,15 @@
-    def initialize(route)
-      @route = route
-      @route_options = route.instance_variable_get(:@options)
+    def initialize(route_options)
```
I bet this class must be initialized with `route` but not `route_options`.
Using `route_options` leads to a cleaner code but violates naming convention.

vaihtovirta committed:
Roll back to previous behavior of GrapeRouteHelpers::DecoratedRoute

- Pass Grape::Route instance to DecoratedRoute
- Roll back tests
- Refactor removing of duplicate routes

vaihtovirta commented:
I've slightly changed my implementation for the sake of naming, now DecoratedRoute is initialized with route instance as it was before. 
Thanks, @timurkhafizov

reprah commented:
Thank you! I'll take a look at your commit very soon. It looks like several people have made a PR about this, I didn't know it was an issue.

reprah commented:
Hi, thanks a lot for taking the time to write a PR and submit it. I chose to merge someone else's fix into the master branch because their change was more contained. Does that make sense? There are things in your pull request that I think are a good idea (like configuring Rspec to run specs in a random order) and will add in.

I really appreciate your help.

reprah commented:
I also released 1.2.2, which should make it work with Ruby 2.3. Please let me know if something else is wrong.

vaihtovirta commented:
@reprah I've checked — it works as expected with 1.2.2 and ruby 2.3.0.
Thank you for release.
